app/models/game.rb

```diff
@@ -57,4 +57,32 @@ def kings_in_check?(color)
     end
     false
   end
+
+  def check_mate?
+    pieces = remaining_pieces
+    checkmate = true
+    friendly = remaining_pieces(color)
+    friendly_moves = friendly.all_moves
+    friendly_moves.each do |move|
+      move
+      checkmate = false unless kings_in_check?
+      undo_move
+      end
+    end
```

Comment:
Is the purpose of this block to see if any of "my" pieces can capture the piece that has me in check?

are `move` and `undo_move` sudo code?